coffea/processor/dataframe.py

```diff
@@ -57,6 +58,7 @@ def __getitem__(self, key):
             if self._flatten and isinstance(awkward1.type(array).type, awkward1.types.ListType):
                 array = awkward1.flatten(array)
             array = awkward1.to_awkward0(array)
```

Comment:
Actually LazyDataFrame could be made awkward1-compatible via opt-in flag that just makes it skip this line. In the long run, though, maybe it makes more sense to support just NanoEvents via BaseSchema

Reply:
Yep - that's why the support warning rather than the will be deleted one. `0.7` here we come.